docs/direct_mig_requirements.md

```diff
@@ -0,0 +1,26 @@
+# Direct Migration Requirements
+
+Direct Migration is available as a feature from Migration Toolkit for Containers (MTC) release 1.4.0. There are two parts of the Direct Migration - Direct 
+Volume Migration and Direct Image Migration. Direct Migration enables the migration of persistent volumes and internal
+images directly from the source cluster to the destination cluster without an intermediary replication repository (object storage)
+There are some requirements which are to be satisfied before you can use the Direct Migration feature of MTC.
+
+1. The clusters (source as well as destination) involved in the migration should expose their respective internal registries
+for external traffic.
+
+2. The remote source and destination clusters should be able to communicate via the OpenShift routes on port 443.
+
+3. The exposed registry route must be configured in the source and destination MigClusters, this can be done by specifying
+the `spec.exposedRegistryPath` field or via the MTC UI.
+
+4. The source as well the destination clusters must be free of all the `Critical` conditions and be in `Ready` state.
+
+5. The Persistent Volumes selected for migration should be valid and exist with a bound state on the source cluster.
+
+6. The Namespaces selected for migration should exist as well as they should be non-empty.
+
+7. The two spec flags in MigPlan CR - `indirectImageMigration` and `indirectVolumeMigration` , both of these flags need to
```

Comment:
What is the default value of these flags?

Reply:
The default value for these flags is `false`

Reply:
That suggests that there is no need to specify these flags are requirements because they would be set to `false` unless the user changed the flags.

Reply:
@apinnick Copying from another comment:

>  FYI, we're still discussing how to maintain backwards compatibility, but it's looking like if these are absent, the migration will use indirect.

>The background on this was that we formerly used direct if these were not set, but the problem with that is it's not backwards compatible with plans created in 1.3.2. Not only did all plans created <1.4.0 become unready upon upgrade, but it also fundamentally changes the semantics with 1.3.2 vs. 1.4.0 (the default behavior would be indirect in 1.3.2, and then suddenly switch to direct with 1.4.0 -- this is not good).

Reply:
@eriknelson I think you have it backwards there: "it's looking like if these are absent, the migration will use indirect." -- if `indirectImageMigration` is absent, that's equivalent to it being false. If `indirectImageMigration` is false, that means *direct* migration will be used. This was the reason the booleans are labeled "indirect" instead of "direct" -- the intent is for direct migration to be the default.

Reply:
This is the first I've heard that it would be bad for the default to switch in 1.4.0. Everything I had heard until now was that we *wanted* the default to change.